pkg/relay/relay.go

```diff
@@ -286,6 +282,10 @@ func (rs *Relay) GetPayload(ctx context.Context, m *structs.MetricGroup, payload
 				logger.Info("published block to beacon node")
 			}
 		}
+
+		if err := rs.d.PutDelivered(ctx, slot, trace, rs.config.TTL); err != nil {
```

Comment:
Just seen it - so, having a parent context this one would be using request's context: 
1) handler running 
2) goroutine spawns 
3) handler finished 
4) Put delivered doesn't really happens, as it would check parent context that is closed already.